codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/HttpProtocolGenerator.kt

```diff
@@ -72,6 +72,14 @@ abstract class HttpProtocolGenerator(
         val builderGenerator = BuilderGenerator(model, symbolProvider, operationShape.inputShape(model))
         builderGenerator.render(inputWriter)
 
+        // TODO: One day, it should be possible for callers to invoke
+        // buildOperationType* directly to get the type rather than depending
+        // on these aliases.
+        val operationTypeOutput = buildOperationTypeOutput(inputWriter, operationShape)
+        val operationTypeRetry = buildOperationTypeRetry(inputWriter, customizations)
+        inputWriter.rust("pub type ${inputShape.id.name}OperationOutputAlias= $operationTypeOutput;")
```

Comment:
do you think we should do doc hidden on these?

Reply:
Yes, absolutely, given that we expect them to go away. Will add!